pkg/scaling/scale_handler.go

```diff
@@ -101,8 +101,20 @@ func (h *scaleHandler) HandleScalableObject(scalableObject interface{}) error {
 
 	// a mutex is used to synchronize scale requests per scalableObject
 	scalingMutex := &sync.Mutex{}
-	go h.startPushScalers(ctx, withTriggers, scalableObject, scalingMutex)
-	go h.startScaleLoop(ctx, withTriggers, scalableObject, scalingMutex)
+	var pullScalableObject interface{}
```

Comment:
Is there a reason that `go h.startPushScalers(ctx, withTriggers, obj.DeepCopy(), scalingMutex)` doesn't work? This seems 💯 on the functionality but very verbose, but I'm guessing you tried the simple one first and this was needed for some weird reason (because Go hates us all)? :D

Reply:
You are absolutely right, I started with something a little bit different and then interated to this verbose version. The simple version is much better, thanks a lot for the review